edge-modules/MessagesAnalyzer/MessagesCache.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace MessagesAnalyzer
+{
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    class MessagesCache
+    {
+        // maps batchId with moduleId, there can be multiple batches for a module
+        readonly ConcurrentDictionary<string, string> batches = new ConcurrentDictionary<string, string>();
```

Comment:
there should be a mapping between the batchId and the moduleId, so that's why dictionary is used.

---
In reply to: [205907753](https://github.com/Azure/iotedge/pull/86#discussion_r205907753) [](ancestors = 205907753)